File: frontend/medops-web/src/app/services/comment.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface CommentDto {
  id: string;
  entityType: string;
  entityId: string;
  userId: string;
  userName: string;
  content: string;
  createdAt: string;
  updatedAt?: string;
}

@Injectable({ providedIn: 'root' })
export class Comment {
  private http = inject(HttpClient);

  getComments(entityType: string, entityId: string): Observable<CommentDto[]> {
    return this.http.get<CommentDto[]>(`/api/comments/${entityType}/${entityId}`);
  }

  addComment(entityType: string, entityId: string, content: string): Observable<CommentDto> {
    return this.http.post<CommentDto>(`/api/comments/${entityType}/${entityId}`, { content });
  }

  updateComment(id: string, content: string): Observable<CommentDto> {
    return this.http.put<CommentDto>(`/api/comments/${id}`, { content });
  }

  deleteComment(id: string): Observable<void> {
    return this.http.delete<void>(`/api/comments/${id}`);
  }
}

File: src/MedOps.Domain/Entities/Comment.cs
namespace MedOps.Domain.Entities;

public class Comment
{
    public Guid Id { get; private set; }
    public string EntityType { get; private set; } = string.Empty;
    public Guid EntityId { get; private set; }
    public Guid UserId { get; private set; }
    public string UserName { get; private set; } = string.Empty;
    public string Content { get; private set; } = string.Empty;
    public DateTime CreatedAt { get; private set; }
    public DateTime? UpdatedAt { get; private set; }
    public bool IsDeleted { get; private set; }

    private Comment() { }

    public Comment(string entityType, Guid entityId, Guid userId, string userName, string content)
    {
        Id = Guid.NewGuid();
        EntityType = entityType;
        EntityId = entityId;
        UserId = userId;
        UserName = userName;
        Content = content;
        CreatedAt = DateTime.UtcNow;
    }

    public void UpdateContent(string content)
    {
        Content = content;
        UpdatedAt = DateTime.UtcNow;
    }

    public void SoftDelete() { IsDeleted = true; UpdatedAt = DateTime.UtcNow; }
}
